application/modules/clients/controllers/Clients.php

```diff
@@ -37,13 +37,40 @@ public function index()
      */
     public function status($status = 'active', $page = 0)
     {
-        if (is_numeric(array_search($status, array('active', 'inactive')))) {
-            $function = 'is_' . $status;
-            $this->mdl_clients->$function();
+        $status_result = array();
+        if ($status == 'all')
+            $status_result = [0,1];
+        elseif ($status == 'active')
+            $status_result = [1];
+        else
+            $status_result = [0];
+
+        $client_results = $this->mdl_clients->with_total_balance_with_currency($status_result);
+
+        //Create an array of number of clients omitting duplicated client in client results
+        //and assigned empty string for client invoice balance to process further.
+        $clients = array();
+
+        foreach ($client_results as $client) {
+            $clients[$client->client_id] = clone $client;
+            $clients[$client->client_id]->client_invoice_balance = '';
         }
 
-        $this->mdl_clients->with_total_balance()->paginate(site_url('clients/status/' . $status), $page);
-        $clients = $this->mdl_clients->result();
+        //Fill the array created above with invoice balance concatenated with total amounts
+        //for a client in different currencies.
+        foreach ($client_results as $client){
+            if (!empty($client->invoice_currency)) {
+                $amount = $client->client_invoice_balance;
+                $symbol = get_currency_symbol($client->invoice_currency);
+
+                if ($amount != 0) {
+                    if ($clients[$client->client_id]->client_invoice_balance)
+                        $clients[$client->client_id]->client_invoice_balance = format_string_type_currency($amount, $symbol, $clients[$client->client_id]->client_invoice_balance);
```

Comment:
I personally am not a fan of this implementation. Currency should be added in another field. There are plugins that rely on the value itself (Zugfred + Sumex) and this will almost certainly break compatibility